src/chia_dialect.rs

```diff
@@ -18,6 +18,10 @@ pub const NO_NEG_DIV: u32 = 0x0001;
 // (otherwise they are no-ops with well defined cost)
 pub const NO_UNKNOWN_OPS: u32 = 0x0002;
 
+// When set, limits the number of atom-bytes allowed to be allocated, as well as
+// the number of pairs
+pub const LIMIT_HEAP: u32 = 0x0004;
```

Comment:
I wanted to keep this patch as small as possible, and not affect existing possible uses of the `new()` constructor. Also, it's non obvious that one limit is `i32::MAX` and one is `i32::MAX - 1` (because we use the negative value space of a signed integer, which is larger)